src/datasets/builder.py

```diff
@@ -174,8 +174,8 @@ def __init__(
         self._cache_dir_root = os.path.expanduser(cache_dir or HF_DATASETS_CACHE)
         self._cache_dir = self._build_cache_dir()
         lock_path = os.path.join(self._cache_dir_root, self._cache_dir.replace("/", "_") + ".lock")
-        with FileLock(lock_path):
-            if os.path.exists(self._cache_dir):  # check if data exist
+        if os.path.exists(self._cache_dir):  # check if data exist
+            with FileLock(lock_path, self._cache_dir):
```

Comment:
Are you sure about this?

`lock_path` is a file inside `self._cache_dir_root`, not inside `self._cache_dir`

Reply:
I was testing something. It's back to what it was